crates/project/src/lsp_store.rs

```diff
@@ -8197,7 +8212,7 @@ impl LspStore {
             .unwrap_or_default();
 
         let new_summary = DiagnosticSummary::new(&diagnostics);
-        if new_summary.is_empty() {
+        if diagnostics.is_empty() {
```

Comment:
In my testing (and to my understanding), this is the only line that needs to be changed to fix the original issue. Is there another case where this is insufficient?